Clarify WebPageProxy wheel event code for UI-side compositing

smfr opened the pull request:
#### 380e2f6bed60dc8af9193a12c8945b8bf1904796
<pre>
Clarify WebPageProxy wheel event code for UI-side compositing
<a href="https://bugs.webkit.org/show_bug.cgi?id=252703">https://bugs.webkit.org/show_bug.cgi?id=252703</a>
rdar://105752960

Reviewed by Tim Horton.

Make use of DrawingAreaProxy::shouldSendWheelEventsToEventDispatcher() to conditionalize
code in WebPageProxy, making it clear that we only call into m_scrollingCoordinatorProxy if we&apos;re
not sending events to the EventDispatcher, and that we don&apos;t need to send ScrollingAccelerationCurves
to the web process with UI-side compositing.

Also dispatchWheelEventWithoutScrolling() is only used on iOS, so #ifdef it for clarity.

* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::handleWheelEvent):
(WebKit::WebPageProxy::sendWheelEvent):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

Canonical link: <a href="https://commits.webkit.org/260752@main">https://commits.webkit.org/260752@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8fb61618657bf74ff6fc0250ea83c15731189fd4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/108955 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/18035 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41779 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/483 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/118246 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19491 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9330 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101186 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114711 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/14623 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97847 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/42743 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/96595 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29488 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/84485 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10817 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30837 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11565 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/7776 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/16953 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50433 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7422 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13161 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -3038,46 +3038,74 @@ void WebPageProxy::flushPendingMouseEventCallbacks()
-    // FIXME: We should not have to look this up repeatedly, but it can also change occasionally.
-    if (event.momentumPhase() == WebWheelEvent::PhaseBegan && preferences().momentumScrollingAnimatorEnabled())
-        m_scrollingAccelerationCurve = ScrollingAccelerationCurve::fromNativeWheelEvent(event);
+        // FIXME: We should not have to look this up repeatedly, but it can also change occasionally.
```
Someone recently told me that there is a way to watch changes to this -- we should consider doing that at some point! This is a whole bunch of work every single scroll.

smfr committed:
Clarify WebPageProxy wheel event code for UI-side compositing
https://bugs.webkit.org/show_bug.cgi?id=252703
rdar://105752960

Reviewed by Tim Horton.

Make use of DrawingAreaProxy::shouldSendWheelEventsToEventDispatcher() to conditionalize
code in WebPageProxy, making it clear that we only call into m_scrollingCoordinatorProxy if we're
not sending events to the EventDispatcher, and that we don't need to send ScrollingAccelerationCurves
to the web process with UI-side compositing.

Also dispatchWheelEventWithoutScrolling() is only used on iOS, so #ifdef it for clarity.

* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::handleWheelEvent):
(WebKit::WebPageProxy::sendWheelEvent):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

Canonical link: https://commits.webkit.org/260752@main
